test/integration/heartbeatAPITest.coffee

```diff
@@ -0,0 +1,147 @@
+should = require 'should'
+request = require 'supertest'
+server = require '../../lib/server'
+Channel = require('../../lib/model/channels').Channel
+Mediator = require('../../lib/model/mediators').Mediator
+testUtils = require '../testUtils'
+auth = require('../testUtils').auth
+
+describe 'API Integration Tests', ->
+  describe 'Heartbeat REST API testing', ->
+
+    mediator1 =
+      urn: 'urn:mediator:awesome-test-mediator'
+      version: '1.0.0'
+      name: 'Awesome Test Mediator'
+      description: 'This is a test mediator. It is awesome.'
+      endpoints: [
+        {
+          name: 'The Endpoint'
+          host: 'localhost'
+          port: '9000'
+          type: 'http'
+        }
+      ]
+
+    authDetails = {}
+
+    before (done) ->
+      auth.setupTestUsers (err) ->
+        return done err if err
+        server.start apiPort: 8080, done
+
+    after (done) ->
+      server.stop -> auth.cleanupTestUsers done
+
+    beforeEach ->
+      authDetails = auth.getAuthDetails()
+
+    afterEach (done) -> Mediator.remove {}, done
+
+    registerMediator = (done) ->
+      request("https://localhost:8080")
+        .post("/mediators")
+        .set("auth-username", testUtils.rootUser.email)
+        .set("auth-ts", authDetails.authTS)
+        .set("auth-salt", authDetails.authSalt)
+        .set("auth-token", authDetails.authToken)
```

Comment:
I think the heartbeats API is public, so I don't think the auth stuff is needed.